helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -113,8 +112,24 @@ public OptimalAssignment calculate(ClusterModel clusterModel) throws HelixRebala
 
     return candidateNodes.parallelStream().map(node -> new HashMap.SimpleEntry<>(node,
         getAssignmentNormalizedScore(node, replica, clusterContext)))
-        .max(Comparator.comparingDouble((scoreEntry) -> scoreEntry.getValue()))
-        .map(Map.Entry::getKey);
+        .max((nodeEntry1, nodeEntry2) -> {
+          int scoreCompareResult =  nodeEntry1.getValue().compareTo(nodeEntry2.getValue());
+          if (scoreCompareResult == 0) {
+            // If the evaluation scores of 2 nodes are the same, the algorithm assigns the replica
+            // to the idle node first.
+            boolean isNodeOneIdle =
```

Comment:
It can be simplified. 
```
int score1 = node1.isBusy() ? 1 : 0 * score1;
int score2 = node2.isBusy() ?  1: 0 * score2;
return score1 - score2;
```

Reply:
That logic will be different, note that the score is still the most important thing. If I change it as you suggested, the busy node will always win regardless of the score. As I explained to you earlier, this should only change the behavior when the evaluation cannot tell the difference between 2 nodes.

Reply:
My suggestion of the multiply is after you check if the score equals.

Reply:
I see. let me have a try.